weaver/store/mongodb.py

```diff
@@ -582,6 +590,25 @@ def find_jobs(self,
         if service is not None:
             search_filters["service"] = service
 
+        if datetime_interval is not None:
+            datetime_interval = datetime_interval_parser(datetime_interval)
+            query = {}
+
+            try:
+                if datetime_interval.get("after", False):
+                    query["$gte"] = datetime_interval["after"]
+
+                if datetime_interval.get("before", False):
+                    query["$lte"] = datetime_interval["before"]
+
+                if datetime_interval.get("match", False):
+                    query = datetime_interval["match"]
```

Comment:
Similar to other comments. 
Function `find_jobs` should take directly the parsed result and not worry about the format of the interval.